openedx/openedx-platform
# feat: show math in plain text in library cards

## Description

Converts mathjax equations to unicode to be rendered as plain text in library card previews.

Useful information to include:

- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author".


**Screenshot:**

![image](https://github.com/user-attachments/assets/b0821ad2-643d-42f9-a672-8beca2a79ee2)


## Supporting information

* https://github.com/openedx/frontend-app-authoring/issues/1508
* `Private-ref`: [FAL-3990](https://tasks.opencraft.com/browse/FAL-3990)

## Testing instructions

* Create some text or problem components with mathjax equations in the body. Make sure to add the equation within first few characters to see the preview in library card.
* Verify that the card shows a plain text preview using unicode.
* Examples of mathjax equations can be found in tests.

## Notes

* [unicodeit](https://github.com/svenkreiss/unicodeit) supports many latex tags but some of them are missing like `\frac`, `\mathbf`, `\bf` etc. which I have handled using replacements.
* Some tags like `\begin{bmatrix}`, `\begin{cases}` etc. cannot be displayed properly in plain text so they are rendered raw. 
* If you come up with some equations which are not rendered properly please let me know.
